ludwig/api.py

```diff
@@ -342,8 +342,8 @@ def train(
             skip_save_processed_input=False,
             output_directory='results',
             gpus=None,
-            gpu_fraction=1.0,
```

Comment:
The presence of gpu_graction was no for reproducibility purposes, but for allowing for the use of a single gpu for multiple trainings. It is used by the hyperopt module quite heavily for instance and is needed together with allow_growth for some problems TF has with RTX cards.

Reply:
Yes, this was intended to be two changes rolled into one:

1. Expose `allow_parallel_threads` to the user, since they had no way to set it previously.
2. Remove `gpu_fraction` because it was removed in TensorFlow 2.

Since there's a need for something like it, we can do this in TensorFlow 2, which is roughly equivalent:

```
gpus = tf.config.experimental.list_physical_devices('GPU')
if gpus:
  try:
    tf.config.experimental.set_logical_device_configuration(gpus[0], [tf.config.experimental.LogicalDeviceConfiguration(memory_limit=1024)])
  except RuntimeError as e:
    print(e)
```

That limits the GPU memory in absolute terms.  If that sounds good, I can add back a parameter like `gpu_memory_limit`.